bzt/modules/aggregator.py

```diff
@@ -243,10 +208,9 @@ def merge_kpis(self, src, sid=None):
             self._concurrencies[sid] = src[self.CONCURRENCY]
 
         if src[self.RESP_TIMES]:
-            # using raw times to calculate percentiles
-            self[self.RESP_TIMES].update(src[self.RESP_TIMES])
-            self.compact_times()
-        elif not self[self.PERCENTILES]:
+            self[self.RESP_TIMES].add(src[self.RESP_TIMES])
+
+        if not self[self.PERCENTILES]:
```

Comment:
`elif` changed to `if`. Won't it change the logic?

Reply:
Agreed, this needs fixing.